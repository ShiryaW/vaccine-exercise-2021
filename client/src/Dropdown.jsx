import React, { Component } from "react";
import { Switch, FormControlLabel } from "@material-ui/core";
import PropTypes from "prop-types";
import { VIEWS } from "./util/constants";

export class Dropdown extends Component {
  constructor(props) {
    super(props);
  }

  render = () => {
    return (
      <div className="dropdown">
        <p>Select view:</p>
        <select id="selectBox" onChange={this.props.onChange}>
          <option value={VIEWS.ANTIQUA.value}>{VIEWS.ANTIQUA.label}</option>
          <option value={VIEWS.SOLARBUDDHICA.value}>
            {VIEWS.SOLARBUDDHICA.label}
          </option>
          <option value={VIEWS.ZERPFY.value}>{VIEWS.ZERPFY.label}</option>
          <option value={VIEWS.VACCINATIONS.value}>
            {VIEWS.VACCINATIONS.label}
          </option>
        </select>
        {!this.props.groupingDisabled ? (
          <FormControlLabel
            label={`Group by ${this.props.groupBy}`}
            labelPlacement="start"
            control={<Switch checked={this.props.buttonChecked} />}
            onChange={this.props.onToggle}
          ></FormControlLabel>
        ) : (
          <></>
        )}
      </div>
    );
  };
}

Dropdown.propTypes = {
  onChange: PropTypes.func.isRequired,
  onToggle: PropTypes.func.isRequired,
  buttonChecked: PropTypes.bool.isRequired,
  groupBy: PropTypes.string.isRequired,
  groupingDisabled: PropTypes.bool,
};
