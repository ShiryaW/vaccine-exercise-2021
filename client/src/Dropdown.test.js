import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import Dropdown from "./Dropdown";
import React from "react";
import { shallow } from "enzyme";

describe("Dropdown", () => {
  beforeEach(() => {
    configure({ adapter: new Adapter() });
  });

  it("renders the top bar with dropdown menu", () => {
    const props = {
      onChange: jest.fn(),
      onToggle: jest.fn(),
      buttonChecked: false,
      groupBy: "gender",
    };
    const dropdown = shallow(<Dropdown {...props} />);
    expect(dropdown).toMatchSnapshot();
  });
});
