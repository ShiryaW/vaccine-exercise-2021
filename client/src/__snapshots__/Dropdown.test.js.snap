// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown renders the top bar with dropdown menu 1`] = `
<div
  className="dropdown"
>
  <p>
    Select view:
  </p>
  <select
    id="selectBox"
    onChange={[MockFunction]}
  >
    <option
      value="antiqua"
    >
      Antiqua orders
    </option>
    <option
      value="solarBuddhica"
    >
      SolarBuddhica orders
    </option>
    <option
      value="zerpfy"
    >
      Zerpfy orders
    </option>
    <option
      value="vaccinations"
    >
      vaccinations
    </option>
  </select>
  <WithStyles(ForwardRef(FormControlLabel))
    control={
      <WithStyles(ForwardRef(Switch))
        checked={false}
      />
    }
    label="Group by gender"
    labelPlacement="start"
    onChange={[MockFunction]}
  />
</div>
`;
